### ISSUE TITLE
Data import: resources

### ISSUE THREAD

**Issue by dpalomino** [Cadasta/cadasta-platform#833]
As a project manager, during the import process I want to upload resources (binary files) that are attached to a record.

**Comment by dpalomino** [comment]
Summarizing some of the ideas discussed with @seav and @bjohare (thanks!!!). 

It is challenging to attach imported resources to entities, as partners are typically going to have a set of many files with different directory structures identifying each party, location, even party name, etc. A typical example would be the case of having multiple resources per record (photographs, GPX files, etc) within a hierarchical directory structure like: Region_name/Village_name/Landowner_name

To develop a solution in the platform for the partners to import all these resources is difficult and quite prone to errors (specially for the non-technologically savvy users). We are suggesting to use a temporary short/medium term solution based on scripts executed by Cadasta staff. The partners that would need to import resource files will not be executing the regular import process (at least in the short term), but we will be supported them. In the longer term we will still to develop a fully integrated process available for the final users. 

So the proposed solution would be: 
- Partner will provide the main CSV import file containing an additional column with a unique "record_id" field. 
- Partner will provide all actual resources/files (within a organized directory structure) directly to Cadasta (this could be done even by courier with USB sticks, to avoid uploading extremely large amounts of data)
- Partner will provide a second CSV file mapping the entity and the resource to be attached. This CSV file will contain the following fields: 
   + The "record_id" identifying the record 
   + A field identifying if the resource is attached to a location, party or relationship. It could be just a column containing "L", "P" or "R".
   + The complete path (directory + file name) of the resource to be attached. We will need to define some minimum rules for the file path names such as unsupported characters or max length. 

Then, the script would follow the next steps, for each record contained in the main CSV file: 
- Create the different entities (with the information contained in the first csv file) using the API
- Using the mapping table contained in the second CSV, it will upload the resources attached to that record to the S3 bucket
- Attach the resource to the proper entity according to the information contained in the mapping table (via API). 

Brian, Eugene, please feel free to add/correct any of the above. Any other thoughts are welcome.

**Comment by wonderchook** [comment]
@dpalomino I don't think it is going to make sense to put requirements on the partners on how they store their data for this. I think we are simply going to have to work with partners on importing their data on a case by case basis. For example how would the partner make a CSV file with the mapping? That seems too complicated for most partners. This is one of the reasons for the technologist position on the programs team so they can help with this sort of work.

**Comment by seav** [comment]
@wonderchook, we actually intend to work with how the partner stores their resource files and not prescribe any structure. But we do need the information on how they map each resource file to each record. We _have_ to assume that the partner has a system for mapping this or otherwise they are just collecting resources uselessly and ineffectively. It would be great if their system is directly translatable into the CSV file that we need, but if their system is algorithmic in nature (they look up a record's resources on-the-fly using some steps instead of consulting some existing index), then the technologist should be able to help the partner construct the mapping CSV file by writing some ad hoc scripts.

**Comment by wonderchook** [comment]
@seav sure I get that, my point was I'm not sure we should do anything from a platform perspective. I think it makes more sense for a script to be written using the API depending on the use case.

**Comment by seav** [comment]
@wonderchook, well, that is actually what's intended for the short/mid-term: the platform will not be modified for importing resources, and we will just create ad hoc scripts or have an easily adaptable set of scripts that we can use to help the partner import resources through the API.

Do you mean that this should also be the case in the long term? That we would not provide an integrated solution for importing resources directly in the platform?

**Comment by wonderchook** [comment]
@seav more I was suggesting we need to wait and see what people's data look likes and then decide what to do in the long term.

**Comment by dpalomino** [comment]
Related to #1436.
